src/bin/consumer/errors.rs

```diff
@@ -13,29 +20,75 @@ impl AssignmentCallbacks for EmptyCallbacks {
     fn on_revoke(&mut self, _: Vec<Partition>) {}
 }
 
+struct Next {
+    destination: TopicOrPartition,
+    producer: KafkaProducer,
+}
+impl ProcessingStrategy<KafkaPayload> for Next {
+    fn poll(&mut self) -> Option<CommitRequest> {
+        None
+    }
+
+    fn submit(&mut self, message: Message<KafkaPayload>) -> Result<(), MessageRejected> {
+        self.producer.produce(&self.destination, &message.payload);
+        Ok(())
+    }
+
+    fn close(&mut self) {}
+
+    fn terminate(&mut self) {}
+
+    fn join(&mut self, _timeout: Option<f64>) -> Option<CommitRequest> {
+        None
+    }
+}
+
+struct StrategyFactory {}
+impl ProcessingStrategyFactory<KafkaPayload> for StrategyFactory {
+    fn create(&self) -> Box<dyn ProcessingStrategy<KafkaPayload>> {
+        let config = KafkaConfig::new_producer_config(vec!["localhost:9092".to_string()], None);
+        let producer = KafkaProducer::new(config);
+        Box::new(Transform {
+            function: identity,
+            next_step: Box::new(Next {
+                destination: TopicOrPartition::Topic({
+                    Topic {
+                        name: "test-dest".to_string(),
+                    }
+                }),
+                producer,
+            }),
+        })
+    }
+}
+
+fn identity(message: Message<KafkaPayload>) -> KafkaPayload {
+    message.payload
+}
+
 fn main() {
     let config = KafkaConfig::new_consumer_config(
         vec!["localhost:9092".to_string()],
         "my_group".to_string(),
-        "latest".to_string(),
+        "earliest".to_string(),
         false,
         None,
     );
-    let mut consumer = KafkaConsumer::new(config);
+    let consumer = KafkaConsumer::new(config);
     let topic = Topic {
-        name: "test_static".to_string(),
+        name: "test-static".to_string(),
     };
-    let res = consumer.subscribe(&[topic], Box::new(EmptyCallbacks {}));
-    assert!(res.is_ok());
+    let mut stream_processor =
```

Comment:
This is annoying but basically consumer.subscribe() takes `&mut self` instead of `&self` because we only actually create the consumer on `subscribe()`. It's this way because of the way the assignment callbacks work right now. rdkafka requires them to be passed when creating the consumer not on the call to subscribe(), but we need to mutate them later to add callbacks in the StreamProcessor.